classes/form/setup_form.php

```diff
@@ -61,13 +61,29 @@ protected function definition() {
         $this->_form->addElement('select', 'weight', get_string('weight', 'block'), $weightoptions);
         $this->_form->addElement('text', 'configdata', get_string('field_configdata', 'tool_blocksmanager'));
         $this->_form->setType('configdata', PARAM_TEXT);
-        $this->_form->addElement('select', 'visible', 'Visible?', [1 => get_string('yes'), 0 => get_string('no')]);
-        $this->_form->addElement('select', 'reposition',  get_string('field_reposition', 'tool_blocksmanager'),
-            [0 => get_string('no'), 1 => get_string('yes')]);
+
+        $this->_form->addElement('select', 'visible', get_string('field_visible', 'tool_blocksmanager'),
+            [1 => get_string('yes'), 0 => get_string('no')]
+        );
+
         $this->_form->addElement('select', 'add', get_string('field_add', 'tool_blocksmanager'),
             [0 => get_string('no'), 1 => get_string('yes')]);
-        $this->_form->hideIf('add', 'reposition', 'eq', 1);
+
+        $this->_form->addElement('select', 'update', get_string('field_update', 'tool_blocksmanager'),
+            [0 => get_string('no'), 1 => get_string('yes')]);
+
+        $this->_form->addElement('select', 'reposition',  get_string('field_reposition', 'tool_blocksmanager'),
+            [0 => get_string('no'), 1 => get_string('yes')]);
+
+        $this->_form->disabledIf('add', 'reposition', 'eq', 1);
+        $this->_form->disabledIf('add', 'update', 'eq', 1);
+
         $this->_form->disabledIf('reposition', 'add', 'eq', 1);
+        $this->_form->disabledIf('reposition', 'update', 'eq', 1);
+
+        $this->_form->disabledIf('update', 'add', 'eq', 1);
```

Comment:
`disabledIf` functions are around anywhere and hard to understand.
I think it's better to group in one place after same element, or same dependency?
e.g.
`$this->_form->disabledIf('update', 'add', 'eq', 1);`  should be immediate after `update` element, or `add` element.

Reply:
fixed